kimchi/src/circuits/polynomials/poseidon.rs

```diff
@@ -137,7 +137,7 @@ impl<F: PrimeField + SquareRootField> CircuitGate<F> {
     /// # Errors
     ///
     /// Will give error if `self.typ` is not `Poseidon` gate, or `state` does not match after `permutation`.
-    pub fn verify_poseidon<G: KimchiCurve<ScalarField = F>>(
+    pub fn verify_poseidon<G: KimchiCurve<ScalarField = F>, const COLUMNS: usize>(
```

Comment:
Making this customisable doesn't make sense while you have `ROUNDS_PER_ROW` hard-coded with `KIMCHI_COLS`. You should either remove this argument, or replace `ROUNDS_PER_ROW` with a helper `const fn rounds_per_row(columns: usize) -> usize`